pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2024,6 +2028,17 @@ func (configgen *ConfigGeneratorImpl) appendListenerFallthroughRouteForCompleteL
 	}
 }
 
+// hash returns a unique hash based on listener.FilterChain, default to 32-bit FNV-1a hash for now
+func hash(fc *listener.FilterChain) (res string, err error) {
+	h := fnv.New32a()
```

Comment:
I have some concerns about the cost and potential non-determinism of the hash. The risk and cost here may outweigh the benefits of getting a name.

Reply:
I share same concern about cost. Can we run some benchmark to see what is the overhead if any?

Reply:
Many thanks for your comments!

IMHO, a benchmark might help but I guess it highly depends on the size of `listener.FilterChain`. And I would agree w/ @howardjohn to some extent, though I think hash can be deterministic if no map-like inputs involved and the cost is OK since it's a general practice.

Besides using a complete standard hash on `listener.FilterChain` for deterministic naming, we have other options like:
1) using a `logical` hash which concatenates some fields within filterchian. However, it's somewhat difficult to decide which fields to pick up as some might be empty; the logical name generated may have implicit semantics which can be confusing; and if we'd like to eliminate that confusion, we'd take another hash anyway.
2) hash on parts of the fields within filterchain. This decreases the cost (though still introduced) but increases the possibility that multi filterchains can get the same name (though seems acceptable for Envoy but loses the intent to be distinct).

Otherwise, if we do have concerns about the cost, can we make this auto-generation configurable similarly to what has been applied in kubectl configmap/secret https://github.com/kubernetes/kubernetes/commit/503a6a8eecd1cbaeceaf54c78ebac14d4d89a4b0?